.storybook/main.js

```diff
@@ -10,9 +11,26 @@ module.exports = {
     );
     svelteLoader.options = {
       ...svelteLoader.options,
-      preprocess: sveltePreprocess({ postcss: true }),
+      preprocess: sveltePreprocess({
+        postcss: true,
+        defaults: {
+          style: "scss",
+        },
+        scss: {
+          prependData: `@import 'src/scss/protocol/includes/_lib.scss';`,
```

Comment:
```suggestion
          prependData: `@import 'node_modules/@mozilla-protocol/core/protocol/css/includes/_lib.scss/scss/protocol/includes/_lib.scss';`,
```

Reply:
The type scale is defined in [/includes/lib](https://github.com/mozilla/glean-dictionary/blob/5e8a4c0f4267b7d195ed43d2eedd6b10ac6c2654/src/scss/protocol/includes/_lib.scss#L43), which then gets the value for sizing from [/includes/themes](https://github.com/mozilla/glean-dictionary/blob/5e8a4c0f4267b7d195ed43d2eedd6b10ac6c2654/src/scss/protocol/includes/_themes.scss#L167). I could just directly update this in node_modules/@mozilla-protocol, but since `node_modules` is gitignored, should I remove all the copied scss files, make the edit directly in the package, and un-gitignore `node_modules/@mozilla-protocol`?

Reply:
Oh I see, no definitely don't un-gitignore anything in `node_modules/*`! That's meant to be installed dynamically at build time.

I think it would be better if we copied as little as possible from protocol into this repository, and reuse as much as possible as-is from what's installed in `node_modules/`. That way we can just upgrade the protocol dependency as we go to gain new features and fixes as they come in, with hopefully a minimal number of changes inside this repository.

However, I admit that I've never done anything like this before (integrating a sass library into a repository) so there might be something I'm missing. I wonder if @craigcook might be able to give us some guidance on how best to handle this (whether that be copying parts of protocol into glean dictionary, or something else...)

Reply:
Yeah right now only what's needed is copied over, which is the /base folder and a couple files from /components. The reason why the number of migrated files is big is because I can't just pick and choose the component files, I have to also copy the dependencies which are the scss variables and mixins in `/includes`. Not sure if it's necessary, but I did add [a README](https://github.com/mozilla/glean-dictionary/blob/use-protocol-and-scss/src/scss/README.md) that explains the process to minimize confusion for other contributors.

Reply:
I'm not sure I see a reason you couldn't copy the whole Protocol framework into your project rather than referencing `node_modules` directly, though I'm not sure how you'd actually do the copying in your particular setup (I've only done it with Gulp). Since you seem to be copying over some of it already you might as well get whole thing.

And because you're now compiling Sass yourself you can definitely pick-and-choose specific components at the compile step. The `protocol-components.css` file is only provided for use if you're **not** compiling your own Sass, but it ends up being a huge file with EVERY component so it will always be more than you need. You don't need it at all if you're compiling yourself, just `@import` the components you're actually using.

What we've done on other sites is copy Protocol from `node_modules` into some local build folder and all our own SCSS is in a local `scss` folder (or whatever you want to name it). The local SCSS can reference the local copy of Protocol files, and any customization or extensions we need to make to it are only done in our own SCSS (relying on specificity or the cascade to override the original Protocol styling). Definitely don't change any of the Protocol source files since any changes would just be overwritten next time you copy it.

You can also configure the brand theme and type scale in your local SCSS.

Reply:
> What we've done on other sites is copy Protocol from `node_modules` into some local build folder and all our own SCSS is in a local `scss` folder (or whatever you want to name it). The local SCSS can reference the local copy of Protocol files, and any customization or extensions we need to make to it are only done in our own SCSS (relying on specificity or the cascade to override the original Protocol styling). Definitely don't change any of the Protocol source files since any changes would just be overwritten next time you copy it.

Is there any reason you'd recommend copying protocol into the project and checking it in, vs. just referencing it via `node_modules`? The latter feels a little more elegant to me.

Reply:
> Is there any reason you'd recommend copying protocol into the project and checking it in, vs. just referencing it via `node_modules`? The latter feels a little more elegant to me.

I suppose not, other than shortening the paths on your imports. I just meant to say you *can* copy it without hurting anything as long as you're not modifying anything.

That's just what we've done on other sites like bedrock or kitsune, which admittedly have a lot more stuff happening during the build process so there might be performance advantages to a local copy in those cases that might not apply to you. And we only copy it into a local build folder; the copy isn't committed to git. 

Some things you _will_ want to copy and commit, such as the fonts, icons, and any scripts you're using from Protocol. Anything that needs to be served to the client, but you could do that manually. All the Protocol SCSS could stay in `node_modules` if that's better for you.